src/shipping/components/PricingCard/PricingCard.tsx

```diff
@@ -75,31 +78,39 @@ export const PricingCard: React.FC<PricingCardProps> = ({
             </TableCell>
           </TableHead>
           <TableBody>
-            {channels?.map(channel => (
-              <TableRow key={channel.id}>
-                <TableCell>
-                  <Typography>{channel.name}</Typography>
-                </TableCell>
-                <TableCell>
-                  <PriceField
-                    disabled={disabled}
-                    label={intl.formatMessage({
-                      defaultMessage: "Price",
-                      description: "column title"
-                    })}
-                    name="price"
-                    value={channel.price}
-                    onChange={e =>
-                      onChange(channel.id, {
-                        ...channel,
-                        price: e.target.value
-                      })
-                    }
-                    currencySymbol={defaultCurrency}
-                  />
-                </TableCell>
-              </TableRow>
-            ))}
+            {channels?.map(channel => {
+              const error = formErrors.price?.find(error =>
+                error.channels?.find(id => id === channel.id)
+              );
+              return (
+                <TableRow key={channel.id}>
+                  <TableCell>
+                    <Typography>{channel.name}</Typography>
+                  </TableCell>
+                  <TableCell>
+                    <PriceField
+                      disabled={disabled}
+                      error={!!error}
+                      label={intl.formatMessage({
+                        defaultMessage: "Price",
+                        description: "column title"
+                      })}
+                      name="price"
+                      value={channel.price}
+                      onChange={e =>
+                        onChange(channel.id, {
+                          ...channel,
+                          price: e.target.value
+                        })
+                      }
+                      currencySymbol={channel.currency}
+                      required
+                      hint={error ? getShippingErrorMessage(error, intl) : ""}
```

Comment:
Since we're doing this fallback to string quite a lot, why don't we default it to `""` in the PriceField component?

Reply:
I wonder if it's necessary at all